File: SCALA/Cap08/c08ex04.scala
// c08ex04.scala

import scala.io.StdIn._

object Main {
  def main(args: Array[String]): Unit = {
    var nomes = new Array[String](5)
    var x: String = ""
    var i, j: Int = 0

    println("CLASSIFICACAO DE NOMES (DECRESCENTE)\n")

    for (i <- 0 to 4) {
      print(s"${ i + 1 }o. nome: ")
      nomes(i) = readLine()
    }

    for (i <- 0 to 3) {
      for (j <- i + 1 to 4) {
        if (nomes(i).compareToIgnoreCase(nomes(j)) > 0) {
          x = nomes(i)
          nomes(i) = nomes(j)
          nomes(j) = x
        }
      }
    }

    println()
    for (i <- 0 to 4) {
      println(s"${ i + 1 }º nome: ${ nomes(i) }")
    }

    print("\nTecle <Enter> para encerrar... ")
    readLine()
  }
}

File: DART/Cap08/c08ex04.dart
// c08ex04.dart

import 'dart:io';

void main() {
  List<String> nomes = List.filled(5, "");
  String x;

  print("CLASSIFICACAO DE NOMES (DECRESCENTE)\n");

  for (int i = 0; i <= 4; i++) {
    stdout.write("${i + 1}o. nome: ");
    nomes[i] = stdin.readLineSync() ?? "";
  }

  for (int i = 0; i <= 3; i++)
    for (int j = i + 1; j <= 4; j++)
      if (nomes[i].compareTo(nomes[j]) > 0) {
        x = nomes[i];
        nomes[i] = nomes[j];
        nomes[j] = x;
      }

  print("");
  for (int i = 0; i <= 4; i++)
    print("${5 - i}o. nome: ${nomes[i]}");

  stdout.write("\nTecle <Enter> para encerrar... ");
  stdin.readLineSync();
}
